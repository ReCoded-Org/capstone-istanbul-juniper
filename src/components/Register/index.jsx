import { Alert, Button, Checkbox, Input } from "antd";
import Modal from "antd/lib/modal/Modal";
import React, { useState } from "react";
import { useTranslation } from "react-i18next";
import { validateEmail } from "../../functions";

const MAX_PASSWORD_LENGTH = 8;

const Register = ({
  onSubmit,
  onGoToLogin,
  onFacebookAuth,
  onGoogleAuth,
  error,
}) => {
  const [t] = useTranslation();
  const [registerInformation, setRegisterInformation] = useState({
    fullName: "",
    age: "",
    email: "",
    password: "",
    agree: false,
  });
  const [errors, setErrors] = useState({});
  const [isTermsOpen, setIsTermsOpen] = useState(false);
  const handleChange = (key, value) => {
    const newValues = { ...registerInformation };
    newValues[key] = value;
    setRegisterInformation(newValues);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    let { fullName, email, age, password, agree } = registerInformation;
    let newErrors = {};
    if (fullName.trim() === "") {
      newErrors["fullName"] = t("register.fullName");
    }
    if (email.trim() === "") {
      newErrors["email"] = t("register.validEmail");
    }
    if (age.trim() === "") {
      newErrors["age"] = t("register.age");
    }
    if (password.trim().length < MAX_PASSWORD_LENGTH) {
      newErrors["password"] = t("register.passwordCharacter");
    }
    if (password.trim() === "") {
      newErrors["password"] = t("register.enterPassword");
    }
    if (!validateEmail(email)) {
      newErrors["email"] = t("register.emailFormat");
    }
    if (!agree) {
      newErrors["agree"] = t("register.agreeOnTerms");
    }
    setErrors(newErrors);
    if (Object.keys(newErrors).length > 0) {
      return;
    }
    onSubmit(registerInformation);
  };

  const handleOkTerms = () => {
    handleChange("agree", true);
    setIsTermsOpen(false);
  };
  const handleCancelTerms = () => {
    setIsTermsOpen(false);
  };
  const termsAndConditionsTranslation = [
    ...t("register.terms", { returnObjects: true }),
  ];

  return (
    <form noValidate onSubmit={handleSubmit}>
      <Modal
        title={t("register.termsTitle")}
        visible={isTermsOpen}
        onOk={handleOkTerms}
        onCancel={handleCancelTerms}
        footer={[
          <Button key="back" onClick={handleCancelTerms}>
            {t("register.close")}
          </Button>,
          <Button key="submit" type="primary" onClick={handleOkTerms}>
            {t("register.iAgree")}
          </Button>,
        ]}
      >
        {t("register.agreeOnTerms", { returnObjects: true })}
      </Modal>
      <div className="loginContainer">
        <div className="loginContainer__loginTitle">{t("register.create")}</div>
        {error && (
          <Alert
            style={{ marginBottom: 10 }}
            type="error"
            showIcon
            message={error}
          ></Alert>
        )}
        <div className="loginContainer__loginDialog">
          <div className="loginContainer__loginDialog__inputLabel">
            {t("register.name")}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Input
              className={
                errors["fullname"] &&
                "loginContainer__loginDialog__input__hasError"
              }
              value={registerInformation.fullname}
              onChange={(e) => {
                handleChange("fullname", e.target.value);
              }}
            />
            {errors["fullname"] && (
              <div className="loginContainer__loginDialog__errorContainer">
                {errors["fullname"]}
              </div>
            )}
          </div>
          <div className="loginContainer__loginDialog__inputLabel">
            {t("register.enterAge")}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Input
              className={
                errors["age"] && "loginContainer__loginDialog__input__hasError"
              }
              type="number"
              value={registerInformation.age}
              onChange={(e) => {
                handleChange("age", e.target.value);
              }}
            />
            {errors["age"] && (
              <div className="loginContainer__loginDialog__errorContainer">
                {errors["age"]}
              </div>
            )}
          </div>
          <div className="loginContainer__loginDialog__inputLabel">
            {t("register.email")}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Input
              className={
                errors["email"] &&
                "loginContainer__loginDialog__input__hasError"
              }
              value={registerInformation.email}
              onChange={(e) => {
                handleChange("email", e.target.value);
              }}
            />
            {errors["email"] && (
              <div className="loginContainer__loginDialog__errorContainer">
                {errors["email"]}
              </div>
            )}
          </div>
          <div className="loginContainer__loginDialog__inputLabel">
            {t("register.password")}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Input
              className={
                errors["password"] &&
                "loginContainer__loginDialog__input__hasError"
              }
              type="password"
              value={registerInformation.password}
              onChange={(e) => {
                handleChange("password", e.target.value);
              }}
            />
            {errors["password"] && (
              <div className="loginContainer__loginDialog__errorContainer">
                {errors["password"]}
              </div>
            )}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Checkbox
              checked={registerInformation.agree}
              onChange={(e) => {
                handleChange("agree", e.target.checked);
              }}
            >
              {t("register.iAgreeTo")}
              <a
                href="#"
                onClick={(e) => {
                  e.preventDefault();
                  setIsTermsOpen(true);
                }}
              >
                <ul>
                  {termsAndConditionsTranslation.map((termAndCondiction) => (
                    <li>{termAndCondiction}</li>
                  ))}
                </ul>
              </a>
            </Checkbox>
            {errors["agree"] && (
              <div className="loginContainer__loginDialog__errorContainer">
                {errors["agree"]}
              </div>
            )}
          </div>
          <button
            type="submit"
            className="loginContainer__loginDialog__submitButton"
          >
            {t("register.createAccount")}
          </button>
          <div className="loginContainer__loginDialog__registerLabel">
            {t("register.alreadyAMember")}
          </div>
          <button
            onClick={onGoToLogin}
            type="button"
            className="loginContainer__loginDialog__registerButton"
          >
            {t("register.login")}
          </button>
          <div className="loginContainer__loginDialog__loginOptionsTitle">
            {t("register.orYouCan")}
          </div>
          <button
            onClick={onFacebookAuth}
            type="button"
            className="loginContainer__loginDialog__facebookLoginBtn"
          >
            {t("register.withFacebook")}
          </button>
          <button
            onClick={onGoogleAuth}
            type="button"
            className="loginContainer__loginDialog__googleLoginBtn"
          >
            {t("register.withGoogle")}
          </button>
        </div>
      </div>
    </form>
  );
};
export default Register;
