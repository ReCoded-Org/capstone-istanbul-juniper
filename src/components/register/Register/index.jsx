import { Alert, Button, Checkbox, Input } from "antd";
import Modal from "antd/lib/modal/Modal";
import React, { useState } from "react";
import { useTranslation } from "react-i18next";
import { validateEmail, hasError } from "../registerHelpers";
import ErrorMessage from "../ErrorMessage";

const Register = ({
  // check container/LoginRegister/index.jsx to see functions
  onSubmit,
  handleLogin,
  handleFacebookAuth,
  handleGoogleAuth,
  // string
  error,
}) => {
  const [t] = useTranslation();
  const [registerInformation, setRegisterInformation] = useState({
    fullName: "",
    age: "",
    email: "",
    password: "",
    isAgreed: false,
  });
  const [errors, setErrors] = useState({});
  const [isTermsOpen, setIsTermsOpen] = useState(false);
  const handleChange = (key, value) => {
    const newValues = { ...registerInformation };
    newValues[key] = value;
    setRegisterInformation(newValues);
  };

  const newErrorsObj = (registerInfo) => {
    const MAX_PASSWORD_LENGTH = 8;
    const { fullName, email, age, password, isAgreed } = registerInfo;
    const errorObj = {};
    if (fullName.trim() === "") {
      errorObj.fullName = t("register.fullName");
    }
    if (age.trim() === "") {
      errorObj.age = t("register.age");
    }
    if (password.trim().length < MAX_PASSWORD_LENGTH) {
      errorObj.password = t("register.passwordCharacter");
    }
    if (password.trim() === "") {
      errorObj.password = t("register.enterPassword");
    }
    if (validateEmail(email).length === 0) {
      errorObj.email = t("register.emailFormat");
    }
    if (!isAgreed) {
      errorObj.isAgreed = t("register.agreeOnTerms");
    }

    return errorObj;
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (Object.keys(newErrorsObj).length > 0) {
      setErrors(newErrorsObj);
      return;
    }
    onSubmit(registerInformation);
  };

  const handleOkTerms = () => {
    handleChange("isAgreed", true);
    setIsTermsOpen(false);
  };
  const handleCancelTerms = () => {
    setIsTermsOpen(false);
  };
  const termsAndConditionsTranslation = [
    ...t("register.terms", { returnObjects: true }),
  ];

  return (
    <form noValidate onSubmit={handleSubmit}>
      <Modal
        title={t("register.termsTitle")}
        visible={isTermsOpen}
        onOk={handleOkTerms}
        onCancel={handleCancelTerms}
        footer={[
          <Button key="back" onClick={handleCancelTerms}>
            {t("register.close")}
          </Button>,
          <Button key="submit" type="primary" onClick={handleOkTerms}>
            {t("register.iAgree")}
          </Button>,
        ]}
      >
        {t("register.agreeOnTerms", { returnObjects: true })}
      </Modal>
      <div className="loginContainer">
        <div className="loginContainer__loginTitle">{t("register.create")}</div>
        {error && (
          <Alert
            style={{ marginBottom: 10 }}
            type="error"
            showIcon
            message={error}
          />
        )}
        <div className="loginContainer__loginDialog">
          <div className="loginContainer__loginDialog__inputLabel">
            {t("register.name")}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Input
              className={hasError(errors.fullName)}
              value={registerInformation.fullName}
              onChange={(e) => {
                handleChange("fullName", e.target.value);
              }}
            />
            <ErrorMessage message={errors.fullName} />
          </div>
          <div className="loginContainer__loginDialog__inputLabel">
            {t("register.enterAge")}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Input
              className={hasError(errors.age)}
              type="number"
              value={registerInformation.age}
              onChange={(e) => {
                handleChange("age", e.target.value);
              }}
            />
            <ErrorMessage message={errors.age} />
          </div>
          <div className="loginContainer__loginDialog__inputLabel">
            {t("register.email")}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Input
              className={hasError(errors.email)}
              value={registerInformation.email}
              onChange={(e) => {
                handleChange("email", e.target.value);
              }}
            />
            <ErrorMessage message={errors.email} />
          </div>
          <div className="loginContainer__loginDialog__inputLabel">
            {t("register.password")}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Input
              className={hasError(errors.password)}
              type="password"
              value={registerInformation.password}
              onChange={(e) => {
                handleChange("password", e.target.value);
              }}
            />
            <ErrorMessage message={errors.password} />
          </div>
          <div className="loginContainer__loginDialog__input">
            <Checkbox
              checked={registerInformation.agree}
              onChange={(e) => {
                handleChange("isAgreed", e.target.checked);
              }}
            >
              {t("register.iAgreeTo")}
              <Button
                type="link"
                onClick={(e) => {
                  e.preventDefault();
                  setIsTermsOpen(true);
                }}
              >
                <ul>
                  {termsAndConditionsTranslation.map((termAndCondition) => (
                    <li key={termAndCondition}>{termAndCondition}</li>
                  ))}
                </ul>
              </Button>
            </Checkbox>
            <ErrorMessage message={errors.isAgreed} />
          </div>
          <button
            type="submit"
            className="loginContainer__loginDialog__submitButton"
          >
            {t("register.createAccount")}
          </button>
          <div className="loginContainer__loginDialog__registerLabel">
            {t("register.alreadyAMember")}
          </div>
          <button
            onClick={handleLogin}
            type="button"
            className="loginContainer__loginDialog__registerButton"
          >
            {t("register.login")}
          </button>
          <div className="loginContainer__loginDialog__loginOptionsTitle">
            {t("register.orYouCan")}
          </div>
          <button
            onClick={handleFacebookAuth}
            type="button"
            className="loginContainer__loginDialog__facebookLoginBtn"
          >
            {t("register.withFacebook")}
          </button>
          <button
            onClick={handleGoogleAuth}
            type="button"
            className="loginContainer__loginDialog__googleLoginBtn"
          >
            {t("register.withGoogle")}
          </button>
        </div>
      </div>
    </form>
  );
};
export default Register;
