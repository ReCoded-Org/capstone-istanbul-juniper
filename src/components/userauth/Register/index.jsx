import { Alert, Button, Checkbox, Input } from "antd";
import Modal from "antd/lib/modal/Modal";
import { useTranslation } from "react-i18next";
import React, { useState } from "react";
import { validateEmail, createErrorClassName } from "../registerHelper";
import ErrorMessage from "../ErrorMessage";

const MIN_PASSWORD_LENGTH = 6;

const Register = ({
  // check containers/LoginRegister/index.jsx to see details of functions
  onSubmit,
  handleLogin,
  handleFacebookAuth,
  handleGoogleAuth,
  // string
  error,
}) => {
  const [t] = useTranslation();
  const [registerInformation, setRegisterInformation] = useState({
    fullName: "",
    age: "",
    email: "",
    password: "",
    isAgreed: false,
  });
  const [errors, setErrors] = useState({});
  const [termsOpen, setTermsOpen] = useState(false);
  const handleChange = (key, value) => {
    const newValues = { ...registerInformation };
    newValues[key] = value;
    setRegisterInformation(newValues);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    const { fullName, email, age, password, isAgreed } = registerInformation;
    const newErrors = {};
    if (fullName.trim() === "") {
      newErrors.fullName = t("register.fullName");
    }
    if (email.trim() === "") {
      newErrors.email = t("register.validEmail");
    }
    if (age.trim() === "") {
      newErrors.age = t("register.age");
    }
    if (password.trim().length < MIN_PASSWORD_LENGTH) {
      newErrors.password = t("register.passwordCharacter");
    }
    if (password.trim() === "") {
      newErrors.password = t("register.enterPassword");
    }
    if (!validateEmail(email)) {
      newErrors.email = t("register.emailFormat");
    }
    if (!isAgreed) {
      newErrors.isAgreed = t("register.agreeOnterms");
    }
    setErrors(errors);
    const errorNameArr = Object.keys(newErrors);
    if (errorNameArr.length > 0) {
      return;
    }
    onSubmit(registerInformation);
  };
  const handleOkTerms = () => {
    handleChange("isAgreed", true);
    setTermsOpen(false);
  };
  const handleCancelTerms = () => {
    setTermsOpen(false);
  };
  const termsAndConditionsTranslation = [
    ...t("register.terms", { returnObjects: true }),
  ];

  return (
    <form noValidate onSubmit={handleSubmit}>
      <Modal
        title={t("register.termsTitle")}
        visible={termsOpen}
        onOk={handleOkTerms}
        onCancel={handleCancelTerms}
        footer={[
          <Button key="back" onClick={handleCancelTerms}>
            {t("register.close")}
          </Button>,
          <Button key="submit" type="primary" onClick={handleOkTerms}>
            {t("register.iAgree")}
          </Button>,
        ]}
      >
        <ul>
          {termsAndConditionsTranslation.map((termAndCondition) => (
            <li key={termAndCondition}>{termAndCondition}</li>
          ))}
        </ul>
      </Modal>
      <div className="loginContainer">
        <div className="loginContainer__loginTitle">{t("register.creat")}</div>
        {error && (
          <Alert className="authAlert" type="error" showIcon message={error} />
        )}
        <div className="loginContainer__loginDialog">
          <div className="loginContainer__loginDialog__inputLabel">
            {t("register.name")}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Input
              className={createErrorClassName(errors.fullName)}
              value={registerInformation.fullName}
              onChange={(e) => {
                handleChange("fullName", e.target.value);
              }}
            />
            <ErrorMessage message={errors.fullName} />
          </div>
          <div className="loginContainer__loginDialog__inputLabel">
            {t("register.enterAge")}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Input
              className={createErrorClassName(errors.age)}
              type="number"
              value={registerInformation.age}
              onChange={(e) => {
                handleChange("age", e.target.value);
              }}
            />
            <ErrorMessage message={errors.age} />
          </div>
          <div className="loginContainer__loginDialog__inputLabel">
            {t("register.email")}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Input
              className={createErrorClassName(errors.email)}
              value={registerInformation.email}
              onChange={(e) => {
                handleChange("email", e.target.value);
              }}
            />
            <ErrorMessage message={errors.email} />
          </div>
          <div className="loginContainer__loginDialog__inputLabel">
            {t("register.password")}
          </div>
          <div className="loginContainer__loginDialog__input">
            <Input
              className={createErrorClassName(errors.password)}
              type="password"
              value={registerInformation.password}
              onChange={(e) => {
                handleChange("password", e.target.value);
              }}
            />
            <ErrorMessage message={errors.password} />
          </div>
          <div className="loginContainer__loginDialog__input">
            <Checkbox
              checked={registerInformation.isAgreed}
              onChange={(e) => {
                handleChange("isAgreed", e.target.checked);
              }}
            >
              {t("register.iAgreeTo")}
              <Button
                type="link"
                onClick={(e) => {
                  e.preventDefault();
                  setTermsOpen(true);
                }}
              >
                {t("register.termsAndConditions")}
              </Button>
            </Checkbox>
            <ErrorMessage message={errors.isAgreed} />
          </div>
          <button
            type="submit"
            className="loginContainer__loginDialog__submitButton"
          >
            {t("register.creatAccount")}
          </button>
          <div className="loginContainer__loginDialog__registerLabel">
            {t("register.alreadyAMember")}
          </div>
          <button
            onClick={(e) => {
              e.preventDefault();
              handleLogin();
            }}
            type="button"
            className="loginContainer__loginDialog__registerButton"
          >
            {t("login.login")}
          </button>
          <div className="loginContainer__loginDialog__loginOptionsTitle">
            {t("login.orYouCan")}
          </div>
          <button
            onClick={(e) => {
              e.preventDefault();
              handleFacebookAuth();
            }}
            type="button"
            className="loginContainer__loginDialog__facebookLoginBtn"
          >
            {t("login.withFacebook")}
          </button>
          <button
            onClick={(e) => {
              e.preventDefault();
              handleGoogleAuth();
            }}
            type="button"
            className="loginContainer__loginDialog__googleLoginBtn"
          >
            {t("login.withGoogle")}
          </button>
        </div>
      </div>
    </form>
  );
};
export default Register;
