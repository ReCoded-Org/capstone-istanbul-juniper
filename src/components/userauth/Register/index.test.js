import React from "react";
import Register from "./index.jsx";
import renderer from "react-test-renderer";

describe("Register Component", () => {
  it("matches the tree", () => {
    const tree = renderer.create(<Register />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
