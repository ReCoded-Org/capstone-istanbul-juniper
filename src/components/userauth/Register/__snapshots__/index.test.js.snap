// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component matches the tree 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="loginContainer"
  >
    <div
      className="loginContainer__loginTitle"
    >
      register.creat
    </div>
    <div
      className="loginContainer__loginDialog"
    >
      <div
        className="loginContainer__loginDialog__inputLabel"
      >
        register.name
      </div>
      <div
        className="loginContainer__loginDialog__input"
      >
        <input
          className="ant-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="loginContainer__loginDialog__inputLabel"
      >
        register.enterAge
      </div>
      <div
        className="loginContainer__loginDialog__input"
      >
        <input
          className="ant-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          type="number"
          value=""
        />
      </div>
      <div
        className="loginContainer__loginDialog__inputLabel"
      >
        register.email
      </div>
      <div
        className="loginContainer__loginDialog__input"
      >
        <input
          className="ant-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="loginContainer__loginDialog__inputLabel"
      >
        register.password
      </div>
      <div
        className="loginContainer__loginDialog__input"
      >
        <input
          className="ant-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          type="password"
          value=""
        />
      </div>
      <div
        className="loginContainer__loginDialog__input"
      >
        <label
          className="ant-checkbox-wrapper"
        >
          <span
            className="ant-checkbox"
            style={Object {}}
          >
            <input
              checked={false}
              className="ant-checkbox-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="checkbox"
            />
            <span
              className="ant-checkbox-inner"
            />
          </span>
          <span>
            register.iAgreeTo
            <button
              className="ant-btn ant-btn-link"
              onClick={[Function]}
              type="button"
            >
              <span>
                register.termsAndConditions
              </span>
            </button>
          </span>
        </label>
      </div>
      <button
        className="loginContainer__loginDialog__submitButton"
        type="submit"
      >
        register.creatAccount
      </button>
      <div
        className="loginContainer__loginDialog__registerLabel"
      >
        register.alreadyAMember
      </div>
      <button
        className="loginContainer__loginDialog__registerButton"
        onClick={[Function]}
        type="button"
      >
        login.login
      </button>
      <div
        className="loginContainer__loginDialog__loginOptionsTitle"
      >
        login.orYouCan
      </div>
      <button
        className="loginContainer__loginDialog__facebookLoginBtn"
        onClick={[Function]}
        type="button"
      >
        login.withFacebook
      </button>
      <button
        className="loginContainer__loginDialog__googleLoginBtn"
        onClick={[Function]}
        type="button"
      >
        login.withGoogle
      </button>
    </div>
  </div>
</form>
`;
